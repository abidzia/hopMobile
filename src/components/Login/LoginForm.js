import React, { Component } from "react";
import {
  View,
  Text,
  TextInput,
  Image,
  StyleSheet,
  ActivityIndicator
} from "react-native";

import { Button } from "../Button";
import Service from "../../utils/service";

/**
 * The LoginForm component of the application.
 * This component is used to show Login Form
 */
export default class LoginForm extends Component {
  state = {
    isLoading: false,
    isAlreadyLoggedIn: false,
    isErrorVisible: false,
    isUserIdError: false,
    isPasswordError: false,
    userId: null,
    password: null
  };

  /**
 * This Methiod is used to Validate login form fields
 */
  validateFields() {
    userId = this.state.userId;
    password = this.state.password;
    userIdValid = true;
    passwordValid = true;
    if (userId == "" || userId == null || userId.length == 0) {
      userIdValid = false;
    }

    if (password == "" || password == null || password.length == 0) {
      passwordValid = false;
    }

    this.setState({
      isUserIdError: !userIdValid,
      isPasswordError: !passwordValid
    });

    return userIdValid && passwordValid;
  }

  /**
   * This Methiod is used to authenticate a user from Services
   */
  authenticate(userId, password) {
    this.setState({ isLoading: true });
    // Calling the Login
    Service.login(userId, password)
      .then(response => {
        if(response.status == "1"){
          this.props.onLogin(true);
          this.setState({ isErrorVisible: false, isLoading: false });
        }
        else{
          this.setState({ isErrorVisible: true, isLoading: false });
          this.props.onLogin(false);
        }
      })
      .catch(error => {
        this.setState({ isErrorVisible: true, isLoading: false });
        this.props.onLogin(false);
      });
  }

  /**
   * This Methiod is called when user clicks on login button
   */
  loginPressed = () => {
    this.setState({ isErrorVisible: false });
    var validation = this.validateFields();
    if (validation) this.authenticate(this.state.userId, this.state.password);
  };

  /**
   * This Methiod is used to get Current login status.
   */
  getLoginStatus() {
    this.setState({ isAlreadyLoggedIn: true });
    Service.getLoginStatus()
      .then(response => {
        this.props.onLogin(true);
        this.setState({ isAlreadyLoggedIn: false });
      })
      .catch(error => {
        this.setState({ isAlreadyLoggedIn: false });
        this.props.onLogin(false);
      });
  }
  // User Validation check
  renderEmptyUserIdError() {
    if (this.state.isUserIdError) {
      return (
        <View style={{ margin: 5 }}>
          <Text style={{ color: "red", fontSize: 16 }}>Field Required</Text>
        </View>
      );
    }
  }
  // Password validation check
  renderEmptyPasswordError() {
    if (this.state.isPasswordError) {
      return (
        <View style={{ margin: 5 }}>
          <Text style={{ color: "red", fontSize: 16 }}>Field Required</Text>
        </View>
      );
    }
  }

  // Show any validation error on login form
  renderAuthenticationError() {
    if (this.state.isErrorVisible) {
      return (
        <View style={{ margin: 20 }}>
          <Text style={{ color: "red", fontSize: 16 }}>
            Invalid User ID or Password
          </Text>
        </View>
      );
    }
  }

  render() {
    return (
        <View style={styles.mainContainer}>
           <Image source={require('../../images/bg.jpg')} style={styles.backgroundImage}/>
            {/*<View
              style={{
                justifyContent: "center",
                alignSelf: "center",
                margin: 10
              }}
            >
              <Image source={require("../../images/logo.medium.png")} />
            </View>*/}
            <View style={styles.container}>
              <TextInput
                style={styles.textInput}
                onChangeText={userId => this.setState({ userId })}
                value={this.state.userId}
                placeholder="User ID"
                placeholderTextColor="grey"
                underlineColorAndroid="transparent"
              />
              {this.renderEmptyUserIdError()}
              <TextInput
                style={styles.textInput}
                onChangeText={password => this.setState({ password })}
                value={this.state.password}
                placeholder="Password"
                placeholderTextColor="grey"
                underlineColorAndroid="transparent"
                secureTextEntry={true}
              />
              {this.renderEmptyPasswordError()}
              {this.renderAuthenticationError()}
                <View style={styles.buttonView}>
                  <View>
                    <Button
                      title="Login"
                      type="primary"
                      elevation="0"
                      onPress={this.loginPressed}
                    />
                  </View>
              </View>
          </View>
          {this.state.isLoading &&
            <ActivityIndicator style={styles.loading} size="large" />}
          {this.state.isAlreadyLoggedIn &&
            <ActivityIndicator style={styles.opaqueLoading} size="large" />}
        </View>
    );
  }
}

const styles = StyleSheet.create({
  mainContainer: {
    flex:1,
    justifyContent: "center",
    alignItems: "center"
  },
  container: {
    justifyContent: "center",
    alignSelf: "stretch",
    padding: 10
  },
  backgroundImage: {
    flex: 1,
    resizeMode: 'stretch', // or 'stretch',
    justifyContent: 'center',
    alignSelf: 'stretch',
    width: undefined,
    height: undefined   
  },
  buttonView: {
    flexDirection: "row",
    justifyContent: "flex-end",
    padding: 10
  },
  textInput: {
    borderBottomColor: "grey",
    borderBottomWidth: 1,
    color: "black",
    fontSize: 16,
    margin: 10
  },
  loading: {
    position: "absolute",
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "rgba(0,0,0,0.1)"
  },
  opaqueLoading: {
    position: "absolute",
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white"
  }
});
