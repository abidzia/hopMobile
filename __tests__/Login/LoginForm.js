import "react-native";
import React from "react";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

import LoginForm from "../../src/components/Login/LoginForm";

it("renders correctly", () => {
  const tree = renderer.create(<LoginForm state={{ isErrorVisible: true }} />);
  expect(tree).toMatchSnapshot();
});
