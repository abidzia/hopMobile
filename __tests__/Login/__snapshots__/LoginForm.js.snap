// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#404040",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
          "padding": 10,
          "width": 500,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="User ID"
        placeholderTextColor="grey"
        style={
          Object {
            "borderBottomColor": "grey",
            "borderBottomWidth": 1,
            "color": "white",
            "fontSize": 16,
            "height": 40,
            "margin": 10,
          }
        }
        underlineColorAndroid="transparent"
        value={null}
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="grey"
        secureTextEntry={true}
        style={
          Object {
            "borderBottomColor": "grey",
            "borderBottomWidth": 1,
            "color": "white",
            "fontSize": 16,
            "height": 40,
            "margin": 10,
          }
        }
        underlineColorAndroid="transparent"
        value={null}
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#f57e20",
              "borderRadius": 3,
              "elevation": 0,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 8,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontFamily": "Roboto",
                  "fontSize": 18,
                  "fontWeight": "bold",
                },
                Object {
                  "color": "white",
                },
                undefined,
              ]
            }
          >
            LOGIN
          </Text>
        </View>
      </View>
    </View>
  </View>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
